drivers/azure/azureutil/cleanup.go

```diff
@@ -189,7 +189,16 @@ func (c *nsgCleanup) CanBeDeleted(ctx context.Context, a AzureClient) bool {
 	sgProp := c.ref.SecurityGroupPropertiesFormat
 	orphaned := (sgProp.Subnets == nil || len(*sgProp.Subnets) == 0) &&
 		(sgProp.NetworkInterfaces == nil || len(*sgProp.NetworkInterfaces) == 0)
-	return orphaned
+	if !orphaned {
+		return false
+	}
+	if c.usedInPool {
+		if _, ok := c.ref.Tags[ManagedByRancherKey]; !ok {
+			log.Warnf("User-provided pooled NSG %s requires manual cleanup or a reassessment of the security rules", c.name)
```

Comment:
What does `or a reassessment of the security rules` mean? If the NSG doesn't have our tag we can just warn that it won't be deleted.

Reply:
This is related to your comment below regarding modifying and existing NSG: 
https://github.com/rancher/machine/pull/77#discussion_r420445351

In order to allow an existing NSG to function properly with Rancher, we need to add a couple of security rules to it:
https://github.com/aiyengar2/machine/blob/27b5a57bfb13353ef2a6dbd39ad9ee7ecfacd07e/drivers/azure/azureutil/securityrules.go#L36-L45

Namely:
- Opening the provided SSH port to all inbound tcp requests (set to 22 in the driver)
- Opening the provided Docker port to all inbound tcp requests (supplied by the user in the picture below)
- Opening all the other requested ports in the `Open Port` section of the Azure node template with the given protocols (shown in the picture below)
![Screen Shot 2020-05-05 at 5 19 42 PM](https://user-images.githubusercontent.com/13127443/81127928-9fa86880-8ef4-11ea-91ac-5bdd07c79e5c.png)

The intent of this message is to notify a user who provides Rancher an NSG that they need to either manually clean up the NSG or, if they intend to continue using it, reassess whether certain security rules need to be deleted.

Reply:
`or a reassessment of the security rules` has been removed since, as of https://github.com/rancher/machine/pull/77#issuecomment-625564555, we no longer need to mutate the NSG provided by a user.